libraries/amqp/include/eosio/amqp/amqp_handler.hpp

```diff
@@ -147,13 +187,18 @@ class amqp {
 
    // called from non-amqp thread
    void wait() {
-      auto r = connected_.get_future().wait_for( std::chrono::seconds( 10 ) );
-      if( r == std::future_status::timeout ) {
-         on_error( "AMQP timeout declaring queue" );
+      auto r = connected_future_.wait_for( std::chrono::seconds( 10 ) );
+      if( r == std::future_status::timeout && on_error_ ) {
+         boost::asio::post( *handler_->amqp_strand(),
+               [on_err = on_error_]() {
+                  try {
+                     on_err( "AMQP timeout declaring queue" );
```

Comment:
This can cause `app().quit()` in `amqp_trace_plugin`'s on_err to be delayed, which allows it to think that initialization succeeded and publish through a null channel.

Reply:
ok, back to using mutex.